p2p/libp2p/netMessenger.go

```diff
@@ -437,16 +466,18 @@ func (netMes *networkMessenger) Broadcast(topic string, buff []byte) {
 
 // RegisterMessageProcessor registers a message process on a topic
 func (netMes *networkMessenger) RegisterMessageProcessor(topic string, handler p2p.MessageProcessor) error {
-	if handler == nil || handler.IsInterfaceNil() {
+	if check.IfNil(handler) {
 		return p2p.ErrNilValidator
 	}
 
 	netMes.mutTopics.Lock()
 	defer netMes.mutTopics.Unlock()
-	validator, found := netMes.topics[topic]
+
+	_, found := netMes.topics[topic]
 	if !found {
 		return p2p.ErrNilTopic
 	}
+	validator := netMes.processors[topic]
 	if validator != nil {
```

Comment:
Is this condition correct? It not should be validator == nil ?

Reply:
The condition is correct, you can not register a topic validator (messageProcessor) on an already set-up topic. (we are not allowing setting more than 1 topic validators on a specific topic)

Reply:
ok